context/app/static/js/components/workspaces/NewWorkspaceDialog/useCreateWorkspaceForm.ts

```diff
@@ -66,6 +67,17 @@ function useCreateWorkspaceForm({ defaultName, defaultTemplate }: UseCreateWorks
     handleClose();
   }
 
+  useEffect(() => {
+    if (defaultProtectedDatasets && defaultProtectedDatasets !== '') {
+      reset({
+        'workspace-name': defaultName ?? '',
+        'protected-datasets': defaultProtectedDatasets ?? '',
+        templates: [defaultTemplate ?? DEFAULT_TEMPLATE_KEY],
+        workspaceJobTypeId: DEFAULT_JOB_TYPE,
+      });
+    }
+  }, [defaultProtectedDatasets, reset, defaultName, defaultTemplate]);
```

Comment:
After looking more into this, you were right this is necessary because the form is created before any selections have been made.

Let's keep this logic, but let's pull the default values e.g. ` defaultName ?? ''` into variables for reuse since they're used both in `defaultValues` and the `useEffect`. Also maybe  'initialProtectedDatasets' instead of 'defaultProtectedDatasets' since they depend on user selections?